conftest.py

```diff
@@ -698,17 +709,39 @@ def pytest_sessionfinish(session, exitstatus):
     session.config.option.log_listener.stop()
 
 
-def is_skip_must_gather(node):
-    all_markers = [mark.name for mark in list(node.iter_markers())]
-    return "skip_must_gather_collection" in all_markers
+def get_all_node_markers(node: Node) -> list[str]:
+    return [mark.name for mark in list(node.iter_markers())]
+
 
+def is_skip_must_gather(node: Node) -> bool:
+    return "skip_must_gather_collection" in get_all_node_markers(node=node)
 
-def pytest_exception_interact(node, call, report):
+
+def get_inspect_command_namespace_string(node: Node, test_name: str) -> str:
+    namespace_str = ""
+    components = [key for key in NAMESPACE_COLLECTION.keys() if f"tests/{key}/" in test_name]
+    if not components:
+        LOGGER.warning(f"{test_name} does not require special data collection on failure")
```

Comment:
should we do return "" here because no further action is needed if not components

Reply:
we return `namespace_str` once after if/else, no need to return it here